src/sql/base/browser/ui/buttonMenu/buttonMenu.ts

```diff
@@ -0,0 +1,319 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./buttonMenu';
+import { Gesture, EventType as GestureEventType } from 'vs/base/browser/touch';
+import { ActionRunner, IAction, IActionRunner } from 'vs/base/common/actions';
+import { BaseActionViewItem, IActionViewItemProvider } from 'vs/base/browser/ui/actionbar/actionbar';
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { AnchorAlignment } from 'vs/base/browser/ui/contextview/contextview';
+import { IMenuOptions } from 'vs/base/browser/ui/menu/menu';
+import { ResolvedKeybinding, KeyCode } from 'vs/base/common/keyCodes';
+import { EventHelper, EventType, removeClass, addClass, append, $, addDisposableListener, addClasses } from 'vs/base/browser/dom';
+import { IContextMenuDelegate } from 'vs/base/browser/contextmenu';
+import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+
+export interface ILabelRenderer {
+	(container: HTMLElement): IDisposable | null;
+}
+
+export interface IBaseDropdownOptions {
+	label?: string;
+	labelRenderer?: ILabelRenderer;
+}
+
+export class BaseDropdown extends ActionRunner {
```

Comment:
Are any of these classes copy/pasted from src/vs/base/browser/ui/dropdown/dropdown.ts?

If so, we shouldn't have these defined again here and should instead reference them.

Reply:
I can clear out all except the class which I had modified.